tests/test.py

```diff
@@ -726,861 +754,102 @@ def test_marathon_objects(
-            hm_state='tests/bigip_test_blank.json'):
-        """Test: Start Marathon app with two service ports."""
-        # Get the test data
-        self.read_test_vectors(cloud_state, bigip_state, hm_state)
-
-        # Do the BIG-IP configuration
-        apps = ctlr.get_apps(self.cloud_data, True)
-        cfg = ctlr.create_config_marathon(self.bigip, apps)
-        self.bigip.regenerate_config_f5(cfg)
-
-        self.check_labels(self.cloud_data, apps)
-
-        # Verify BIG-IP configuration
-        self.assertFalse(self.bigip.pool_update.called)
-        self.assertFalse(self.bigip.healthcheck_update.called)
-        self.assertFalse(self.bigip.member_update.called)
-        self.assertFalse(self.bigip.virtual_update.called)
-        self.assertFalse(self.bigip.iapp_update.called)
-
-        self.assertFalse(self.bigip.ltm.virtuals.virtual.load.called)
-        self.assertFalse(self.bigip.ltm.pools.get_collection.called)
-        self.assertFalse(self.bigip.ltm.monitor.https.http.load.called)
-        self.assertFalse(self.bigip.ltm.monitor.tcps.tcp.load.called)
-        self.assertFalse(self.bigip.member_delete.called)
-        self.assertFalse(self.bigip.iapp_delete.called)
-        self.assertFalse(self.bigip.iapp_create.called)
-
-        self.assertTrue(self.bigip.member_create.called)
-        self.assertEquals(self.bigip.member_create.call_count, 6)
-
-        expected_name1 = 'server-app4_8080'
-        expected_name2 = 'server-app4_8090'
-        self.assertEquals(2, len(self.test_virtual))
-        self.assertEquals(2, len(self.test_pool))
-        self.assertEquals(2, len(self.test_monitor))
-        self.assertEquals(expected_name1, self.test_virtual[0]['name'])
-        self.assertEquals(expected_name2, self.test_virtual[1]['name'])
-        self.assertEquals(self.test_partition,
-                          self.test_virtual[0]['partition'])
-        self.assertEquals(self.test_partition,
-                          self.test_virtual[1]['partition'])
-        self.assertEquals(expected_name1, self.test_pool[0]['name'])
-        self.assertEquals(expected_name2, self.test_pool[1]['name'])
-        self.assertEquals(self.test_partition, self.test_pool[0]['partition'])
-        self.assertEquals(self.test_partition, self.test_pool[1]['partition'])
-        self.assertEquals(expected_name1, self.test_monitor[0]['name'])
-        self.assertEquals(expected_name2, self.test_monitor[1]['name'])
-        self.assertEquals(self.test_partition,
-                          self.test_monitor[0]['partition'])
-        self.assertEquals(self.test_partition,
-                          self.test_monitor[1]['partition'])
+    def test_marathon_configs(self):
+        """Test: Verify expected BIG-IP config created from Marathon state."""
+        # Verify configuration
+        for key, value in marathon_test_data.items():
+            self.verify_marathon_config(key, value)
 
     def start_two_apps_with_two_partitions(
```

Comment:
_two_partitions to _multiple_partitions since it is used with more than 2 partitions.